bin/propolis-server/src/lib/spec/config_toml.rs

```diff
@@ -94,7 +89,24 @@ impl TryFrom<&config::Config> for ParsedConfig {
     type Error = ConfigTomlError;
 
     fn try_from(config: &config::Config) -> Result<Self, Self::Error> {
-        let mut parsed = Self::default();
+        let mut parsed = ParsedConfig {
+            enable_pcie: config
+                .chipset
+                .options
+                .get("enable-pcie")
+                .map_or_else(
+                    || Ok(false),
+                    |v| {
+                        v.as_bool().ok_or_else(|| {
+                            ConfigTomlError::EnablePcieParseFailed(
+                                v.to_string(),
+                            )
+                        })
+                    },
+                )?,
```

Comment:
_whew_

Reply:
(personally, i find `map_or_else` _way_ harder to interpret than `.map(...).or_else(...)`, but...i know Clippy will probably yell at you for that...)

Reply:
I played this a little differently (do the `ok_or_else` inside the first `map` and then use `transpose` to extract that result) but have thus removed the `map_or_else` and trimmed the indentation here.